diff --git bazel/cython_library.bzl bazel/cython_library.bzl

--- bazel/cython_library.bzl
+++ bazel/cython_library.bzl
@@ -19,7 +19,7 @@
 # been written at cython/cython and tensorflow/tensorflow. We branch from
 # Tensorflow's version as it is more actively maintained and works for gRPC
 # Python's needs.
-def pyx_library(name, deps = [], py_deps = [], srcs = [], **kwargs):
+def pyx_library(name, deps = [], cc_kwargs = {}, py_deps = [], srcs = [], **kwargs):
     """Compiles a group of .pyx / .pxd / .py files.
 
     First runs Cython to create .cpp files for each input .pyx or .py + .pxd
@@ -31,6 +31,7 @@ def pyx_library(name, deps = [], py_deps = [], srcs = [], **kwargs):
     Args:
         name: Name for the rule.
         deps: C/C++ dependencies of the Cython (e.g. Numpy headers).
+        cc_kwargs: cc_binary extra arguments such as copts, linkstatic, linkopts, features
         py_deps: Pure Python dependencies of the final library.
         srcs: .py, .pyx, or .pxd files to either compile or pass through.
         **kwargs: Extra keyword arguments passed to the py_library.
@@ -70,11 +71,12 @@ def pyx_library(name, deps = [], py_deps = [], srcs = [], **kwargs):
         stem = src.split(".")[0]
         shared_object_name = stem + ".so"
         native.cc_binary(
-            name = shared_object_name,
-            srcs = [stem + ".cpp"],
-            deps = deps + ["@local_config_python//:python_headers"],
+            name = cc_kwargs.pop("name", shared_object_name),
+            srcs = [stem + ".cpp"] + cc_kwargs.pop("srcs", []),
+            deps = deps + ["@local_config_python//:python_headers"] + cc_kwargs.pop("deps", []),
             defines = defines,
-            linkshared = 1,
+            linkshared = cc_kwargs.pop("linkshared", 1),
+            **cc_kwargs
         )
         shared_objects.append(shared_object_name)
 
